addons/l10n_pe_website_sale/static/src/js/website_sale.js

```diff
@@ -0,0 +1,90 @@
+/** @odoo-module **/
+import {WebsiteSale} from "@website_sale/js/website_sale";
+
+WebsiteSale.include({
+    events: Object.assign({}, WebsiteSale.prototype.events, {
+        "change select[name='state_id']": "_onChangeState",
+        "change select[name='city_id']": "_onChangeCity",
+    }),
+    start: function () {
+        this.selectCities = document.querySelector("select[name='city_id']");
+        this.selectDistricts = document.querySelector("select[name='l10n_pe_district']");
+        this.cityBlock = document.querySelector(".div_city");
+        this.autoFormat = document.querySelector(".checkout_autoformat");
+        this.selectState = document.querySelector("select[name='state_id']");
+        this.selectCountry = document.querySelector("select[name='country_id']");
+        this.isPeruvianCompany = this.selectCountry?.dataset.company_country_code === 'PE';
+        return this._super.apply(this, arguments);
+    },
+    _changeOption: function (selectCheck, rpcRoute, place, selectElement) {
+        if (!selectCheck) {
+            return;
+        }
+        return this.rpc(rpcRoute, {
+        }).then((data) => {
+            if (this.isPeruvianCompany) {
+                if (data[place]?.length) {
+                    selectElement.innerHTML = "";
+                    data[place].forEach((item) => {
+                        let opt = document.createElement("option");
+                        opt.textContent = item[1];
+                        opt.value = item[0];
+                        opt.setAttribute("data-code", item[2]);
+                        selectElement.appendChild(opt);
+                    });
+                    selectElement.parentElement.style.display = "block";
+                } else {
+                    selectElement.value = "";
+                    selectElement.parentElement.style.display = "none";
+                }
+            }
+        });
+    },
+    _onChangeState: function () {
+        if (this.isPeruvianCompany) {
+            if (this.selectState.value === "" && this.selectCountry.value !== '') {
+                this.selectState.options[1].selected = true;
+            }
+            const state = this.selectState.value;
+            const rpcRoute = `/shop/state_infos/${state}`;
+            return this.autoFormat.length
+                ? this._changeOption(state, rpcRoute, "cities", this.selectCities).then(() => this._onChangeCity())
+                : undefined;
+        }
+    },
+    _onChangeCity: function () {
+        if (this.isPeruvianCompany) {
+            const city = this.selectCities.value;
+            const rpcRoute = `/shop/city_infos/${city}`;
+            return this.autoFormat.length
```

Comment:
I'm not really sure what this `autoFormat` does

Reply:
@h4818 

is not a method it's a selector https://github.com/odoo/odoo/pull/130155/files#diff-3e73cdaf8077b89c48579b44b997cff31cf049b58fc66ed417209e307f5b0de4R13

it just check if it exists in the DOM with the `.length` property.

cc @luistorresm